src/mat4.js

```diff
@@ -1226,6 +1226,65 @@ export function getRotation(out, mat) {
   return out;
 }
 
+/**
+ * Decomposes a transformation matrix into its rotation, translation
+ * and scale components. Returns only the rotation component
+ * @param  {quat} out_r Quaternion to receive the rotation component
+ * @param  {vec3} out_t Vector to receive the translation vector
+ * @param  {vec3} out_s Vector to receive the scaling factor
+ * @param  {ReadonlyMat4} mat Matrix to be decomposed (input)
+ * @returns {quat} out_r
+ */
+export function decompose(out_r, out_t, out_s, mat) {
+  getTranslation(out_t, mat);
+  getScaling(out_s, mat);
```

Comment:
Looks good! Though, one minor performance improvement would be to manually inline those two functions.